tests/unit/workflows/test_workflows_tasks_arxiv.py

```diff
@@ -35,59 +35,96 @@
 from inspirehep.modules.workflows.tasks.arxiv import (
     arxiv_author_list,
     arxiv_derive_inspire_categories,
-    arxiv_fulltext_download,
     arxiv_package_download,
     arxiv_plot_extract,
+    populate_arxiv_document,
 )
 from plotextractor.errors import InvalidTarball
 
 from mocks import AttrDict, MockEng, MockFiles, MockObj
 
 
-def test_arxiv_fulltext_download_logs_on_success():
-    with requests_mock.Mocker() as requests_mocker:
-        requests_mocker.register_uri(
-            'GET', 'http://export.arxiv.org/pdf/1605.03844',
-            content=pkg_resources.resource_string(
-                __name__, os.path.join('fixtures', '1605.03844.pdf')),
-        )
+def test_populate_arxiv_document():
+    schema = load_schema('hep')
+    subschema = schema['properties']['arxiv_eprints']
 
-        schema = load_schema('hep')
-        subschema = schema['properties']['arxiv_eprints']
+    data = {
+        'arxiv_eprints': [
+            {
+                'categories': [
+                    'physics.ins-det',
+                ],
+                'value': '1605.03844',
+            },
+        ],
+    }  # literature/1458302
 
-        data = {
-            'arxiv_eprints': [
-                {
-                    'categories': [
-                        'physics.ins-det',
-                    ],
-                    'value': '1605.03844',
-                },
-            ],
-        }  # literature/1458302
-        extra_data = {}
-        files = MockFiles({})
-        assert validate(data['arxiv_eprints'], subschema) is None
+    assert validate(data['arxiv_eprints'], subschema) is None
+    extra_data = {}
+    files = MockFiles({})
+    obj = MockObj(data, extra_data, files=files)
+    eng = MockEng()
 
-        obj = MockObj(data, extra_data, files=files)
-        eng = MockEng()
+    assert populate_arxiv_document(obj, eng) is None
+    expected_url = 'http://export.arxiv.org/pdf/1605.03844'
+    expected_document = {
+        'key': '1605.03844.pdf',
+        'fulltext': True,
+        'hidden': True,
+        'material': 'preprint',
+        'original_url': expected_url,
+        'url': expected_url,
+    }
+    documents = obj.data['documents']
+    assert 1 == len(documents)
+    assert expected_document['url'] == documents[0]['url']
```

Comment:
It might be better to improve the assertion in these cases, something to compare the whole object WDTY?

Reply:
@drjova aha, good catch forget about this 😄